### ISSUE TITLE
"yarn add --offline " Couldn't find any versions

### ISSUE THREAD

**Issue by huangyingjie** [yarnpkg/yarn#1221]
**yarn version**

 0.16.0

**Report a *bug**

**What is the current behavior?**

$  yarn cache ls
````js
yarn cache v0.16.0
Name   Version Registry Resolved
lodash 4.16.4  npm      http://r.npm.sankuai.com/lodash/download/lodash-4.16.4.tgz#01ce306b9bad1319f2a5528674f88297aeb70127
✨  Done in 0.09s.
````
$ yarn add --offline lodash@4.16.4
````js
yarn add v0.16.0
warning No license field
[1/4] 🔍  Resolving packages...
error Couldn't find any versions for "lodash" that matches "4.16.4" in our cache. Possible versions: ""
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
````
**If the current behavior is a bug, please provide the steps to reproduce.**
* yarn add lodash
* yarn add --offline lodash@4.16.4

**What is the expected behavior?**
no error.
**Please mention your node.js, yarn and operating system version.**
node v4.4.1
os: macOSX EI 10.11.6

**Comment by verth** [comment]
I believe it to have been a small oversight during a refactor. The `metadata._remote` check in the [resolveRequestOffline](https://github.com/yarnpkg/yarn/blob/master/src/resolvers/registries/npm-resolver.js#L104) is always falsy as it resolves as `remote` ([readPackageMetadata](https://github.com/yarnpkg/yarn/blob/master/src/config.js#L315)). 

In the initial `--offline` [commit](https://github.com/yarnpkg/yarn/commit/7d5da38400cf1d7a9f3381c75c4d301c243a39f6) you can see it still uses `remote` over `_remote`, it got changed to `_remote` in a [commit](https://github.com/yarnpkg/yarn/commit/986938e9242b852dcc9e7488f6a34abf627174a9) with a lot of naming refactors. Other occurrences that use `readPackageMetadata` also still use `remote`.

**Comment by kittens** [comment]
Fixed via #1277.
